packages/react/src/components/ComposedModal/ModalHeader.tsx

```diff
@@ -0,0 +1,204 @@
+}
+
+export const ModalHeader = React.forwardRef<HTMLDivElement, ModalHeaderProps>(
+  function ModalHeader(
+    {
+      buttonOnClick,
+      children,
+      className: customClassName,
+      closeClassName,
+      closeIconClassName,
+      closeModal,
+      iconDescription,
+      label,
+      labelClassName,
+      title,
+      titleClassName,
+      ...rest
+    },
+    ref
+  ) {
+    const prefix = usePrefix();
+
+    function handleCloseButtonClick(evt: MouseEvent) {
+      closeModal?.(evt);
+      buttonOnClick?.(evt);
+    }
+
+    const headerClass = cx(`${prefix}--modal-header`, customClassName);
+
+    const labelClass = cx(
+      `${prefix}--modal-header__label ${prefix}--type-delta`,
+      labelClassName
+    );
+
+    const titleClass = cx(
+      `${prefix}--modal-header__heading ${prefix}--type-beta`,
+      titleClassName
+    );
+
+    const closeClass = cx(`${prefix}--modal-close`, closeClassName);
+
+    const closeIconClass = cx(
+      `${prefix}--modal-close__icon`,
+      closeIconClassName
+    );
+
+    return (
+      <div className={headerClass} {...rest} ref={ref}>
+        {label && <h2 className={labelClass}>{label}</h2>}
+
+        {title && <h3 className={titleClass}>{title}</h3>}
+
+        {children}
+
+        <button
+          onClick={handleCloseButtonClick}
+          className={closeClass}
+          title={iconDescription}
+          aria-label={iconDescription}
+          type="button">
+          <Close size={20} className={closeIconClass} />
+        </button>
+      </div>
+    );
+  }
+);
+
+ModalHeader.propTypes = {
+  /**
+   * Provide an optional function to be called when the close button is
+   * clicked
+   */
+  buttonOnClick: PropTypes.func,
+
+  /**
+   * Specify the content to be placed in the ModalHeader
+   */
+  children: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the modal header
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify an optional className to be applied to the modal close node
+   */
+  closeClassName: PropTypes.string,
+
+  /**
+   * Specify an optional className to be applied to the modal close icon node
+   */
+  closeIconClassName: PropTypes.string,
+
+  /**
+   * Provide an optional function to be called when the modal is closed
+   */
+  closeModal: PropTypes.func,
+
+  /**
+   * Specify a description for the close icon that can be read by screen
+   * readers
+   */
+  iconDescription: PropTypes.string,
+
+  /**
+   * Specify an optional label to be displayed
+   */
+  label: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the modal header label
+   */
+  labelClassName: PropTypes.string,
+
+  /**
+   * Specify an optional title to be displayed
+   */
+  title: PropTypes.node,
+
+  /**
+   * Specify an optional className to be applied to the modal heading
+   */
+  titleClassName: PropTypes.string,
+};
+
+ModalHeader.defaultProps = {
+  iconDescription: 'Close',
+};
```

Comment:
what happened to the `buttonOnClick` & `closeModal` default props here?
<img width="253" alt="image" src="https://user-images.githubusercontent.com/40550942/233739994-243a916c-c320-4ef1-a53d-11b0636d1a7a.png">